ansible/roles_ocp_workloads/bookbag/tasks/workload.yaml

```diff
@@ -0,0 +1,68 @@
+---
+# Implement your Workload deployment tasks here
+
+- name: Create bookbag namespace
+  k8s:
+    api_key: "{{ bookbag_openshift_token | default(omit) }}"
+    host: "{{ bookbag_openshift_api_url | default(omit) }}"
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: "{{ bookbag_namespace }}"
+
+- name: Create temporary directory for bookbag source
+  tempfile:
+    prefix: bookbag-{{ guid }}-
+    state: directory
+  register: r_bookbag_tmp
+
+- name: Clone bookbag repository to output dir
+  git:
+    repo: "{{ bookbag_git_repo }}"
+    version: "{{ bookbag_git_version | default(omit) }}"
+    dest: "{{ r_bookbag_tmp.path }}"
+
+- name: Process bookbag build template
+  command: >-
+    oc process --local -f {{ (r_bookbag_tmp.path ~ '/build-template.yaml') | quote }} -o json
```

Comment:
Is there no way of doing this with k8s and yaml templates?

Reply:
It could be done with yaml templates, but the way we had folks do bookbag development was using OpenShift templates we gave them for build and deploy. I did it this way to match what the development process.